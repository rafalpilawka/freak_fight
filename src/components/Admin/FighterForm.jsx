import { withFormik } from "formik";
import React from "react";
import styled from 'styled-components'
import { Form, FormGroup, Label, Button } from 'reactstrap';

const FormGroupContainer = styled.div`
	 {
		display: flex;
		width: 100%;
		flex-direction: column;
		align-items: center;
		padding: 10px;
		flex-grow: 1;
		flex-basis: 1;
		text-transform: uppercase;
		font-family: 'Teko', sans-serif;
		font-size: 1.5rem;
	}
`;

const FighterContainer = styled.div`
	 {
		display: flex;
		flex-direction: column;
		align-items: center;
		background-color: black;
		color: white;
		margin: 25px;
		padding: 30px;
		border-radius: 5px;
		width: 100%;
	}
`;

const Input = styled.input`
	font-size: 16px;
	border: solid 1px #dbdbdb;
	border-radius: 3px;
	color: #262626;
	padding: 7px 33px;
	border-radius: 3px;
	color: #999;
	cursor: text;
	font-size: 1.5rem;
	font-weight: 300;
	text-align: center;
	background: #fafafa;
	&:active,
	&:focus {
		text-align: left;
		outline: none !important;
		border: 1px solid red;
		box-shadow: 0 0 10px red;
	}
`;
function FighterForm(props) {
  const { values, handleChange, onChange } = props;
  // console.log(values)
  React.useEffect(() => {
    if (onChange) {
      onChange(values);
    }
  }, [values]);

  return (
    <div className="form">
      <FighterContainer className="fighterContainer">
        <FormGroupContainer className="formgroupcontainer">
          <Label for="nick">Fighter nick.</Label>
          <Input
            type="text"
            name="nick"
            id="nick"
            value={values.nick}
            onChange={handleChange}
            required
          />
        </FormGroupContainer>
        <FormGroupContainer>
          <Label for="name">values name.</Label>
          <Input
            type="text"
            name="name"
            id="name"
            value={values.name}
            onChange={handleChange}
            required
          />
        </FormGroupContainer>
        <FormGroupContainer>
          <Label for="fighter1Photo">Fighter weight.</Label>
          <Input
            type="text"
            name="weight"
            id="weight"
            value={values.weight}
            onChange={handleChange}
          />
        </FormGroupContainer>
        <FormGroupContainer>
          <Label for="height">Fighter height.</Label>
          <Input
            type="text"
            name="height"
            id="height"
            value={values.height}
            onChange={handleChange}
          />
        </FormGroupContainer>
        <FormGroupContainer>
          <Label for="last_fight">Fighter last fight.</Label>
          <Input
            type="text"
            name="last_fight"
            id="last_fight"
            value={values.last_fight}
            onChange={handleChange}
          />
        </FormGroupContainer>
        <FormGroupContainer>
          <Label for="fighterIMg">Fighter photo.</Label>
          <Input
            type="text"
            name="fighterImg"
            id="fighterImg"
            value={values.fighterImg}
            onChange={handleChange}
          />
        </FormGroupContainer>
      </FighterContainer>
    </div>

  );
}

export default withFormik({
  mapPropsToValues: () => {
    return {
      name: '',
      nick: '',
      weight: '',
      height: '',
      last_fight: '',
      fighterImg: ''
    };
  }
})(FighterForm);




