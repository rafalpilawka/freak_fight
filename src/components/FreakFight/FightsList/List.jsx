import React from 'react'
import Fight from 'components/FreakFight/FightsList/Fight/Fight';

const List = ({fights, votedMatrix, fireModal}) => {
  console.log("TCL: List -> votedMatrix", votedMatrix)
  return fights.map((fight, i) =>
    <Fight modal={fireModal} id={fight.id} key={fight.id} fight={fight} voted={!votedMatrix ? false : votedMatrix[i]} />
  )
}

export default List
