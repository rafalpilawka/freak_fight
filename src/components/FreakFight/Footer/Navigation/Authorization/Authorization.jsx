import styled from 'styled-components';
import React, { useState, useContext, useEffect } from 'react';
import {useFirebase} from 'Firebase/FreakFight/index';
import {ReactComponent as FBlogo} from 'components/FreakFight/Footer/Navigation/Authorization/fb_ico.svg'

const LoginStatusContainer = styled.div`
	 {
		display: flex;
		align-items: center;
		justify-content: center;
		justify-items: center;
		cursor: pointer;
		font-family: 'Roboto Mono', monospace;
		text-transform: uppercase;
		cursor: pointer;
	}
`;

const BtnFacebookContainer = styled.button`
    ${'' /* width: 135px; */}
		width: 135px;
    height:35px; 
    border-radius: 4px;
    background: #3b5998;
    color:white;
    border:0px transparent;  
    text-align: center;
    display: flex;
		justify-items: space-between;
		align-items: center;
		cursor: ponter;
		font-weight: bold;
		font-size: 14px;
		padding: 0px;
		padding-top: 0px;

    &:hover{
        background: #3b5998;
        opacity: 0.6;
    }
		& .fb{
			width: 20%;
			padding: 0;
			margin: 0px;
			padding-left: 5%;
		} 
		& p{
			height: 100%;
			width: 100%;
			margin: 0px;
			z-index: 999;
		}

		@media only screen and (max-width : 650px) {
        width: 135px;
				height: 45px;
				font-size: 17px;
				justify-items: space-between;
				& p{
					position: relative;
					text-align: left;
				}
    }
`;

const AuthorizationContainer = () => {
	const { auth, doSignInWithFacebook, doSignOut, addUserToDB } = useFirebase();
	const [userAuth, setUserAuth] = useState(() => auth.currentUser);

	useEffect(() => {
		const authVar = auth.onAuthStateChanged(function(user) {
			if (user) {
				setUserAuth(true);
			} else {
				setUserAuth(false);
			}
		});
	});

	const signInWithFacebook = () => {
		if (!auth.currentUser) {
			doSignInWithFacebook()
		} else {
			doSignOut()
				.then(res => setUserAuth(false))
				.catch(err => console.log(err));
		}
	};

	return (
		<LoginStatusContainer 
		onClick={signInWithFacebook}
		>
				{userAuth ? 
				<BtnFacebookContainer className='btnface' >
					<FBlogo className='fb'/>
					<p>&nbsp;&nbsp;Wyloguj się</p>
				</BtnFacebookContainer >
				: <BtnFacebookContainer className='btnface'>
				<FBlogo className='fb' />
				<p>&nbsp;&nbsp;Zaloguj się</p>
				</BtnFacebookContainer >}
		</LoginStatusContainer>
	);
};

export default AuthorizationContainer;
