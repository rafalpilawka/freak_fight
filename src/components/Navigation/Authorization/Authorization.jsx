import React, { useContext, useState } from 'react';
import {UserContext} from 'context/userContext'
import styled from 'styled-components'
import FirebaseContext from 'firebase/context';

const Button = styled.button`
{

}`
const LoginStatusContainer = styled.div`{
    display: flex;
    align-items: center;
    justify-content: center;

}`

const AuthorizationContainer = (props) => {


    console.log(props)
    const { userStatus, setUserStatus } = useContext(UserContext);
    const { auth, doSignInWithFacebook, doSignOut} = useContext(FirebaseContext)

    const signInWithFacebook=()=>{

        if(!auth.currentUser){
            doSignInWithFacebook().then(socialAuthUser => setUserStatus({ userStatus: auth.currentUser })).catch(err => console.log(err))
        }else{
            doSignOut().then(res =>{ console.log(res);setUserStatus({ userStatus: false})}).catch(err=>console.log(err))
        }
    }
    
        return (
            <LoginStatusContainer>
                        <Button onClick={signInWithFacebook} >
                    {auth.currentUser ? 'You are signedIn' : 'Log in' }
                        </Button>
            </LoginStatusContainer>
        );
}

export default AuthorizationContainer;