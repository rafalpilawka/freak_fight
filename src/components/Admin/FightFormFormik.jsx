import React, { useState } from 'react';
import FighterForm from 'components/Admin/FighterForm';
import styled from 'styled-components';
import { useFirebase } from 'Firebase/FreakFight/index';

const FormContainer = styled.div`
	 {
		align-items: center;
		justify-content: space-between;
		heigth: 500px;
		width: 100%;
		margin-top: 20px;
	}
`;

const ButtonContainer = styled.button`
	 {
		padding: 8px;
		font-size: 1.5rem;
		text-transform: uppercase;
		font-family: 'Teko', sans-serif;
		color: white;
		background-color: black;
		border-radius: 5px;
		position: relative;
		&:hover {
			background-color: darkred;
			color: #fdec6e;
			box-shadow: 0px 0px 5px #fdec6e;
			-moz-transition: all 0.2s ease-in;
			-o-transition: all 0.2s ease-in;
			-webkit-transition: all 0.2s ease-in;
			transition: all 0.1s ease-in;
		}
	}
`;

const FightersContainer = styled.div`
	 {
		display: grid;
		grid-template-columns: 1fr 1fr;
		grid-column-gap: 2em;
		justify-items: center;
		& .submit_button {
			grid-column: 1/3;
			margin: 15px;
		}
	}
`;

const ItemContainer = styled.div`
	 {
		justify-self: stretch;
		justify-content: center;
		align-items: center;
		& p {
			text-align: center;
			justify-content: center;
			padding: 0px !important;
			color: white;
			heigth: 100%;
			font-size: 2em;
		}
	}
`;

const FightFormFormik = () => {
	const initialState = {
		favFighter1: [],
		favFighter2: [],
		fighter1: {
			name: '',
			nick: '',
			weight: '',
			height: '',
			last_fight: '',
			fighterImg: ''
		},
		fighter2: {
			name: '',
			nick: '',
			weight: '',
			height: '',
			last_fight: '',
			fighterImg: ''
		},
		votesForFav: [],
		votesForWin: [],
		winFighter1: [],
		winFighter2: []
	};

	const [formValues, setFormValues] = useState(initialState);
	const { addFighterHandler } = useFirebase();
	const [reset, setReset] = useState(false);

	function handleFormAChange(values) {
		setFormValues({
			...formValues,
			fighter1: values
		});
	}

	function handleFormBChange(values) {
		setFormValues({
			...formValues,
			fighter2: values
		});
	}

	const handleSubmit = () => {
		addFighterHandler(formValues);
		setFormValues({ ...initialState });
		setReset(!reset);
	};

	return (
		<FormContainer>
			<FightersContainer>
				<ItemContainer>
					<p data-testid="fighter">FIGHTER ONE</p>
				</ItemContainer>
				<ItemContainer>
					<p>FIGHTER TWO</p>
				</ItemContainer>
				<ItemContainer>
					<FighterForm onChange={handleFormAChange} reset={reset} />
				</ItemContainer>
				<ItemContainer>
					<FighterForm onChange={handleFormBChange} reset={reset} />
				</ItemContainer>
				<ButtonContainer className="submit_button" onClick={handleSubmit}>
					Add Fight
				</ButtonContainer>
			</FightersContainer>
		</FormContainer>
	);
}

export default FightFormFormik;
