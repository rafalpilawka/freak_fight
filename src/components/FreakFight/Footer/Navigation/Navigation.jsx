import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { ReactComponent as Logo } from 'components/FreakFight/Footer/Navigation/freakfight_logo22.svg';
import Authorization from 'components/FreakFight/Footer/Navigation/Authorization/Authorization';
import { useFirebase } from 'Firebase/FreakFight/index';

const Nav = styled.div`
{
	display: flex;
	justify-content: space-between;
	align-content: center;
	background-color: black;
	color: white;
	width: 100%;
	@media only screen and (max-width: 700px) {
		align-content: flex-start;
	}
}
`;
const LogoStyle = styled.div`
{
padding: 20px;
}
`;
const NavigationLinks = styled.div`
{
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: space-between;
	color: white ;
	padding-right: 30px;
	font-size: 1em;
	text-transform: uppercase;

	@media only screen and (max-width: 700px) {
		display: none;
	}
}
`;

const NavlinkPad = styled.div`
{
padding: 15px;

a{
	text-decoration: none;
	color: white;
	}
}
`;

const Navigation = () => {
	console.log(useFirebase)
	const { auth } = useFirebase();
	return (
		<Nav className='nav_bar'>
				<LogoStyle>
					<Link className='linklogo' to='/'><Logo className='logo' /></Link>
				</LogoStyle>
				<NavigationLinks>
						<NavlinkPad>
								<Link to="/archives">Arch</Link>
						</NavlinkPad>
						<NavlinkPad>
								<Link to="/heroes">Heroes</Link>
						</NavlinkPad>
						<Authorization isAuth={auth.currentUser} />
				</NavigationLinks>
		</Nav>
	);
};

export default Navigation;
