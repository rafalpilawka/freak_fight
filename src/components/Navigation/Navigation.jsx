import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { ReactComponent as Logo } from './freakfight_logo22.svg';
import Authorization from 'components/Navigation/Authorization/Authorization';
import FirebaseContext from 'firebase/context';

const Nav = styled.div`
{
	display: flex;
	justify-content: space-between;
	align-content: center;
	height: 80px;
	background-color: black;
	color: white;
	width: 100%;
}
`;
const LogoStyle = styled.div`
{
	padding: 10px;
}
`;
const NavigationLinks = styled.div`
{
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: space-between;
	color: white !important;
	padding-right: 10px;
}
`;

const NavlinkPad = styled.div`
{
padding: 15px;

a{
	text-decoration: none;
	color: white;
	}
}
`;

const Navigation = () => {
	const { auth } = useContext(FirebaseContext);

	return (
		<Nav>
				<LogoStyle>
					<Link to='/'><Logo /></Link>
				</LogoStyle>
				<NavigationLinks>
						<NavlinkPad>
							<Link to ='/admin'>Admin</Link>
						</NavlinkPad>
						<NavlinkPad>
								<Link to="/archives">Arch</Link>
						</NavlinkPad>
						<NavlinkPad>
								<Link to="/heroes">Heros</Link>
						</NavlinkPad>
						<Authorization isAuth={auth.currentUser} />
				</NavigationLinks>
		</Nav>
	);
};

export default Navigation;
