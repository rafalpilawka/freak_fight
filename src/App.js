import React, { useState, useRef } from 'react';
import 'App.css';
import {
	BrowserRouter as Router,
	Route,
	Switch,
} from 'react-router-dom';
import Navigation from 'components/FreakFight/Footer/Navigation/Navigation';
import FightsList from 'components/FreakFight/FightsList/FightsList';
import styled from 'styled-components';
import Footer from 'components/FreakFight/Footer/Footer';
import Admin from 'components/Admin/Admin';
import {Burger , Menu} from 'components/FreakFight/Footer/Navigation/Menu/Menu'

const RootAppStyle = styled.div`{
	display: grid;
	grid-template-rows: 80px auto 40px;
	background-color: black;
	grid-template-columns: auto;
	font-size: 6px;
	@media only screen and (min-width: 400px ) {
		font-size: 8px;
	}
	@media only screen and (min-width: 850px) {
		font-size: 18px;
	}
}
`;

function App() {
	const [open, setOpen] = React.useState(false);
	const node = React.useRef();
	return (
		<div className="App">
			<Router>
				<RootAppStyle>
				<Navigation />
				<Switch>
					<Route exact path="/" component={FightsList} />
					<Route exact path="/heroes" />
					<Route exact path='/admin' component={Admin} />
					<Route exact path="/archive" />
				</Switch>
				<Footer />
				<div ref={node}>
					<Burger open={open} setOpen={setOpen} />
					<Menu open={open} setOpen={setOpen} />
				</div>
				</RootAppStyle>
			</Router>
		</div>
	);
}

export default App;
