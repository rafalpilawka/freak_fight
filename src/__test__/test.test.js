import React from 'react';
import {
	render,
	fireEvent,
	waitForElement,
	wait,
	cleanup
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import styled from 'styled-components';
import FightFormFormik from 'components/Admin/FightFormFormik';
import FighterForm from 'components/Admin/FighterForm';
import App from 'App.js';
// import * as Firebase from 'firebase'
import { useFirebase, Provider } from 'Firebase/FreakFight/index';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { JestEnvironment } from '@jest/environment';
import List from 'components/FreakFight/FightsList/List';
import Navigation from 'components/Navigation/Navigation';
import Authorization from 'components/Navigation/Authorization/Authorization';

afterAll(cleanup);
beforeEach(() => {
	jest.resetModules();
});

const renderComponent = ({ component }) => {
	return render(
		<Provider>
			{component}
		</Provider>
	);
};

test('should render navigation ', () => {
	// const {queryByText, getByText} = renderComponent({category : value})
	const { getByText, getByTestId, container } = renderComponent({
		component: <App />
	});
	expect(getByTestId('freakfight_logo')).toBeInTheDocument();
	expect(getByText('KTO WYGRA?')).toBeInTheDocument();
	// expect(getAllByText('Kto wygra')).toContain('Zaloguj się');
});

test('shouldDisplay Form', async () => {
	const {
		getByText,
		getAllByLabelText,
		container,
		asFragment
	} = renderComponent({ component: <FightFormFormik /> });
	const forms = getAllByLabelText(/fighter Nick./i);
	const form = forms[0];
	const vakluesArray = ['raf', 'pil', 'norman'];
	const submit = getByText(/Add fight/i);

	vakluesArray.forEach(item => {
		wait(fireEvent.change(form, { target: { value: item } }));
		wait(expect(form).toHaveValue(item));
		// expect(asFragment()).toMatchSnapshot();
		// fireEvent.click(submit)
		// wait(() => expect(form).toHaveValue(''))
	});
	// fireEvent.focus(form);
});

test('presence of logo in nav compoennt', () => {
	const { getByText, getByTestId, container } = renderComponent({
		component: <App />
	});
	const logo = getByTestId(/freakfight_logo/i);
	expect(logo).toBeTruthy();
});

// fights, votedMatrix, fireModal
// RENDER FIGHTS LIST FROM DATA
test('renders', () => {
	const fights = [
		{
			id: '1y1glBaCdagflxUL6485',
			favFighter1: [],
			favFighter2: [],
			fighter1: {
				fighterImg:
					'https://famemma.com/app/uploads/2019/10/marcin-malczynski.png',
				height: '',
				last_fight: '',
				name: '',
				nick: 'Malczyński ',
				weight: ''
			},
			fighter2: {
				fighterImg:
					'https://famemma.com/app/uploads/2019/10/marek-hoffmann-adbuster.png',
				height: '',
				last_fight: '',
				name: '',
				nick: 'Ad buster',
				weight: ''
			},
			votesForFav: [],
			votesForWin: [
				'0skW1rqHbXhPUCaUfU7JKs0QsFW2',
				'dskJPwCpTOOjn81tViVwZldlkhR2'
			],
			winFighter1: [],
			winFighter2: [
				'0skW1rqHbXhPUCaUfU7JKs0QsFW2',
				'dskJPwCpTOOjn81tViVwZldlkhR2'
			]
		},
		{
			id: 'n332MW4XpxtFiwG10di3',
			favFighter1: [],
			favFighter2: [],
			fighter1: {
				fighterImg: 'https://famemma.com/app/uploads/2019/10/mini-majk.png',
				height: '',
				last_fight: '',
				name: '',
				nick: 'Mini majk',
				weight: ''
			},
			fighter2: {
				fighterImg: 'https://famemma.com/app/uploads/2019/10/lord-kruszwil.png',
				height: '',
				last_fight: '',
				name: '',
				nick: 'Lord Kruszwil',
				weight: ''
			},
			votesForFav: [],
			votesForWin: [
				'0skW1rqHbXhPUCaUfU7JKs0QsFW2',
				'dskJPwCpTOOjn81tViVwZldlkhR2'
			],
			winFighter1: [],
			winFighter2: [
				'0skW1rqHbXhPUCaUfU7JKs0QsFW2',
				'dskJPwCpTOOjn81tViVwZldlkhR2'
			]
		},
		{
			id: 'vABkwXA63NlIDc3LVBE8',
			favFighter1: [],
			favFighter2: [],
			fighter1: {
				fighterImg: 'https://famemma.com/app/uploads/2019/10/bonus.png',
				height: '',
				last_fight: '',
				name: '',
				nick: 'BONUS BGC',
				weight: ''
			},
			fighter2: {
				fighterImg: 'https://famemma.com/app/uploads/2019/10/najman.png',
				height: '',
				last_fight: '',
				name: '',
				nick: 'Najman',
				weight: ''
			},
			votesForFav: [],
			votesForWin: [
				'0skW1rqHbXhPUCaUfU7JKs0QsFW2',
				'dskJPwCpTOOjn81tViVwZldlkhR2'
			],
			winFighter1: [],
			winFighter2: [
				'0skW1rqHbXhPUCaUfU7JKs0QsFW2',
				'dskJPwCpTOOjn81tViVwZldlkhR2'
			]
		}
	];
	const { asFragment } = render(
		<Provider>
			<List fights={fights} votedMatrix={[]} fireModal={() => false} />
		</Provider>
	);
	expect(asFragment()).toMatchSnapshot();
});

test('insert text in h1', () => {
	const { getByTestId, getByText } = render(
		<h1 className="h1tag" data-testid="h1tag">
			Text
		</h1>
	);
	expect(getByTestId('h1tag')).toHaveTextContent('Text');
	expect(getByText('Text')).toHaveClass('h1tag');
});

test('render navigation depends on auth status', () => {
	const { getByText, getByTestId } = renderComponent({
		component: (
			<Router>
				<Navigation />
			</Router>
		)
	});

	expect(getByTestId('heroes_main')).toHaveTextContent('Heroes');
});

test('authorization test', () => {
	// Firebase.doSignInWithFacebook({ currentUser: { greeting: "hello there" } });
	const { getByText, getByTestId, queryByText } = renderComponent({
		component: <Authorization />
	});
	const LogButton = getByTestId(/auth_button/i);
	expect(LogButton).toHaveTextContent;
	expect(getByText(/Zaloguj się/i)).toBeInTheDocument();
	// console.log("TCL: LogButton", LogButton)

	// wait(fireEvent.click(LogButton));
	// const logStatus = await waitForElement(() => getByText('Wyloguj się'))
	// expect(queryByText(/Wyloguj się/i)).toBeInTheDocument()
});

// toHaveTextContent = check for containnig string
