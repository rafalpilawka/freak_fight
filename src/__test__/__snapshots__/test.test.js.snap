// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="sc-kgoBCf rUmIk"
  >
    <div
      class="sc-VigVT iRjkhm fighter-wrapper fighter1grid"
    >
      <div
        class="sc-jTzLTM gIzLzv"
      >
        <img
          class="sc-fjdhpX jSgrqQ"
          src="https://famemma.com/app/uploads/2019/10/marcin-malczynski.png"
        />
        <div
          class="sc-cSHVUG jsioyC winFighter"
        >
          <button
            class="sc-kAzzGY dYejuL"
          />
        </div>
        <div
          class="sc-jzJRlG bkIspK"
        >
          Malczy≈Ñski 
        </div>
      </div>
    </div>
    <div
      class="sc-kGXeez fdFUfe"
    >
      <svg
        styles="[object Object]"
      >
        vslogo.svg
      </svg>
    </div>
    <div
      class="sc-VigVT bsTZTI fighter-wrapper fighter2grid"
    >
      <div
        class="sc-jTzLTM gIzLzv"
      >
        <img
          class="sc-fjdhpX jSgrqQ"
          src="https://famemma.com/app/uploads/2019/10/marek-hoffmann-adbuster.png"
        />
        <div
          class="sc-cSHVUG jsioyC winFighter"
        >
          <button
            class="sc-kAzzGY dYejuL"
          />
        </div>
        <div
          class="sc-jzJRlG bkIspK"
        >
          Ad buster
        </div>
      </div>
    </div>
  </div>
  <div
    class="sc-kgoBCf rUmIk"
  >
    <div
      class="sc-VigVT iRjkhm fighter-wrapper fighter1grid"
    >
      <div
        class="sc-jTzLTM gIzLzv"
      >
        <img
          class="sc-fjdhpX jSgrqQ"
          src="https://famemma.com/app/uploads/2019/10/mini-majk.png"
        />
        <div
          class="sc-cSHVUG jsioyC winFighter"
        >
          <button
            class="sc-kAzzGY dYejuL"
          />
        </div>
        <div
          class="sc-jzJRlG bkIspK"
        >
          Mini majk
        </div>
      </div>
    </div>
    <div
      class="sc-kGXeez fdFUfe"
    >
      <svg
        styles="[object Object]"
      >
        vslogo.svg
      </svg>
    </div>
    <div
      class="sc-VigVT bsTZTI fighter-wrapper fighter2grid"
    >
      <div
        class="sc-jTzLTM gIzLzv"
      >
        <img
          class="sc-fjdhpX jSgrqQ"
          src="https://famemma.com/app/uploads/2019/10/lord-kruszwil.png"
        />
        <div
          class="sc-cSHVUG jsioyC winFighter"
        >
          <button
            class="sc-kAzzGY dYejuL"
          />
        </div>
        <div
          class="sc-jzJRlG bkIspK"
        >
          Lord Kruszwil
        </div>
      </div>
    </div>
  </div>
  <div
    class="sc-kgoBCf rUmIk"
  >
    <div
      class="sc-VigVT iRjkhm fighter-wrapper fighter1grid"
    >
      <div
        class="sc-jTzLTM gIzLzv"
      >
        <img
          class="sc-fjdhpX jSgrqQ"
          src="https://famemma.com/app/uploads/2019/10/bonus.png"
        />
        <div
          class="sc-cSHVUG jsioyC winFighter"
        >
          <button
            class="sc-kAzzGY dYejuL"
          />
        </div>
        <div
          class="sc-jzJRlG bkIspK"
        >
          BONUS BGC
        </div>
      </div>
    </div>
    <div
      class="sc-kGXeez fdFUfe"
    >
      <svg
        styles="[object Object]"
      >
        vslogo.svg
      </svg>
    </div>
    <div
      class="sc-VigVT bsTZTI fighter-wrapper fighter2grid"
    >
      <div
        class="sc-jTzLTM gIzLzv"
      >
        <img
          class="sc-fjdhpX jSgrqQ"
          src="https://famemma.com/app/uploads/2019/10/najman.png"
        />
        <div
          class="sc-cSHVUG jsioyC winFighter"
        >
          <button
            class="sc-kAzzGY dYejuL"
          />
        </div>
        <div
          class="sc-jzJRlG bkIspK"
        >
          Najman
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
